### PR TITLE
Teacher entity added in feature2

### PR DESCRIPTION
**Pull request by shikha1108**
please review it

### CODE REVIEW

File: src/main/java/com/schoolmanagement/school/controller/TeacherController.java [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+import java.util.List;
+import java.util.Optional;
+
+@RestController
+public class TeacherController {
+    @Autowired
+    private TeacherService teacherService;
+
+    @Autowired
+    private TeacherRepository teacherRepository;
```
Comment (pankaj2806) [inline src/main/java/com/schoolmanagement/school/controller/TeacherController.java:20]: We we need repository in Controller?

File: src/main/java/com/schoolmanagement/school/controller/TeacherController.java [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+    }
+
+    @DeleteMapping(value = "/teacher/{id}")
+    public void deleteTeacher(@PathVariable("id") Long id) {
+        teacherService.deleteTeacher(id);
+    }
+
+    @PatchMapping(value = "/teacher/all")
+    public ResponseEntity<Object> updatePartiallyData(@PathVariable("id") Long id, @Validated @RequestBody Teacher teacher) {
+        Optional<Teacher> teacherFromDB = teacherRepository.findById(id);
```
Comment (pankaj2806) [inline src/main/java/com/schoolmanagement/school/controller/TeacherController.java:39]: We should always use service in controller.

File: src/main/java/com/schoolmanagement/school/entity/Teacher.java [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+public class Teacher {
+    @Id
+    @GeneratedValue(strategy = GenerationType.IDENTITY)
+    Long id;
+
+    @Column(name = "first_Name")
+    String firstName;
+
+    @Column(name = "last_Name")
+    String lastName;
```
Comment (pankaj2806) [inline src/main/java/com/schoolmanagement/school/entity/Teacher.java]: Make these fields private.

File: src/main/java/com/schoolmanagement/school/controller/TeacherController.java (hunk 1 above)
Comment (shikha1108) [reply src/main/java/com/schoolmanagement/school/controller/TeacherController.java:20]: Here I m using repository in patch mapping annotation

**Pull request merged**